src/swap/components/SwapSettingsSlippageInput.tsx

```diff
@@ -0,0 +1,123 @@
+import { useCallback, useState } from 'react';
+import { background, border, cn, color, pressable } from '../../styles/theme';
+import type { SwapSettingsSlippageInputReact } from '../types';
+import { useSwapContext } from './SwapProvider';
+
+const slippageSettings = ['Auto', 'Custom'];
+
+export function SwapSettingsSlippageInput({
+  className,
+  defaultSlippage = 3,
+}: SwapSettingsSlippageInputReact) {
+  const { setLifeCycleStatus } = useSwapContext();
+  const [slippage, setSlippage] = useState(defaultSlippage);
+  const [isAutoSlippageSetting, setIsAutoSlippageSetting] = useState(true);
+
+  const updateSlippage = useCallback(
+    (newSlippage: number) => {
+      setSlippage(newSlippage);
+      setLifeCycleStatus({
+        statusName: 'slippageChange',
+        statusData: { maxSlippage: newSlippage },
+      });
+    },
+    [setLifeCycleStatus],
+  );
+
+  // Handles user input for custom slippage
+  // Parses the input and updates slippage if valid
+  const handleSlippageChange = useCallback(
+    (newSlippage: string) => {
+      const newSlippageNumber = Number.parseFloat(newSlippage);
+      if (!Number.isNaN(newSlippageNumber)) {
+        updateSlippage(newSlippageNumber);
+      }
+    },
+    [updateSlippage],
+  );
+
+  // Toggles between auto and custom slippage settings
+  // Resets to default slippage when auto is selected
+  const handleSlippageSettingChange = useCallback(
+    (auto: boolean) => {
```

Comment:
Next PR, always have a boolean starting as either `is` or `has`, as I was reading `auto` and I was questioning what was that?

Reply:
It is true that types helps specify what variable do, but naming helps create intution on what things are even before checking the title.

Like I read this
```
if (auto) {
        updateSlippage(defaultSlippage);
}
```
and I think, is `auto` a car? is `auto` a person? what the heck is `auto`?

Reply:
I agree! `Auto` is a bad name. It should be updated to something like `default`

Reply:
I'll bring this up tomorrow. I believe design is using "`Auto`" because that's the terminology that Uniswap uses.